jeancochrane/philly-fliers
# Sketch out primary app components

# Overview

Defines the primary components for the app and brings the functionality up to the same level as the [Grout blueprint](https://github.com/azavea/grout-blueprint).

## Notes

- This repo doesn't have any unit tests yet. It felt a little premature to me to write them, since I've been writing the code so quickly and doing a lot of rewriting as I go. Let me know if that feels right to you, and/or when you would expect me to start writing tests.

- Most of the action here is in `app/src/`. Here's a quick tour of the directory:

```
src/
├── api.js                 # Configures a Grout API instance
├── App.vue                # Main Vue app is defined here
├── assets/
├── components/
│   ├── Content.vue        # Loads the main content for the Index view (map + sidebar)
│   ├── filters
│   │   ├── Container.vue  # Dispatches filters based on type
│   │   └── Text.vue       # Text filter
│   ├── Filters.vue        # Loads all filters
│   ├── Footer.vue
│   ├── Index.vue          # Main page (view for the `/` route)
│   ├── Map.vue
│   ├── Modal.vue
│   ├── Navbar.vue
│   └── Sidebar.vue
├── Grout.js               # Grout JavaScript API (could become a separate module)
├── main.js                # App entrypoint
├── router
│   └── index.js           # Vue routes
└── store.js               # Vuex datastore
```

## Testing

You can build the app with the standard scripting workflow:

- `./scripts/update` builds containers, runs migrations, loads sample data, and installs NPM modules (hopefully eslint will no longer steal your NPM auth token)

- `./scripts/server` serves:
    - the frontend app on port 8080
    - the Grout API server on port 8000
    - the schema editor on port 9000

- You should be able to:
    1. View records on the map
    2. Click on map markers to get a JSON blob showing the record data
    3. Switch RecordTypes with the dropdown menu
    4. Filter records with the text filter boxes
